### DISCUSSION TITLE
Seeking help with a simple use of Kymatio to generate coefficients

### DISCUSSION THREAD

**Discussion by hopper333** [kymatio/kymatio#1056] [Q&A]
Hello to everyone

First of all, apologies for asking very simple questions here but I could not find answers on the main Kymatio website, help pages or examples.

I have become very interested in the theory behind wavelet scattering transforms and I want to experiment on a simple data set such as CIFAR-10 or CIFAR-100.  I do not want to go as far as a classification problem, just to generate a set of coefficients for each image in the training and test sets. 

I have installed Python 3.12 and SciPy 1.11.4 as suggested on the website and so I am good to go in terms of basic Python support. My problem is that I have very little experience with Python, NumPy, SciPy or Scikit-learn (all I need as I am working on CPUs only at the moment), despite having a long working history of developing in compiled languages such as various flavours of C.

I have gone through the website and looked at/downloaded all the 2D examples (https://www.kymat.io/gallery_2d/index.html) to try and understand the code, but I'm really not clear what is doing what and how much of it is required for just calculating and outputting the coefficients.  I am also not clear about the choices needed to create different types and quantities of coefficients, though understanding the papers better will I think help to clarify this.  So a summary of the help that I need is examples that show the following:

1.     Reading the CIFAR-* into the appropriate data structure for generating the coefficients.
2.     Choosing a sensible set of convolution types/operators/channels/strides/angles/other(?) that will produce 'a low hundred to a few thousand' coefficients for each 32x32 RGB image in the data set.  Are there any guidelines for this?
3.     The function calls that generate the coefficients from the data set stored in memory.
4.     Storing these coefficients and then outputting to a binary or ASCII file - preferably as an 8- or 16-bit integer and normalised in some sensible way.

If there are any links, user guides, tech reports or simple code examples that could help me to get a grasp on the issues then that would be excellent.  What I am hoping to end up with is a data set of 60k rows with all the coefficients for each image in the row.  Ideally, these coefficients would be normalised in some way so that they have the same minimum and maximum possible values.

If you can help in any way that would be greatly appreciated. Thanks in advance

Michael
